src/apis/hermez-web.ts

```diff
@@ -1,17 +1,31 @@
 import axios from "axios";
+import * as z from "zod";
 
 import { HERMEZ_WEB_URL } from "src/constants";
 
 const baseApiUrl = HERMEZ_WEB_URL;
 
+interface WalletUnderMaintenanceResponse {
+  isWalletUnderMaintenance: number;
+}
+
+const walletUnderMaintenanceResponseParser: z.ZodSchema<WalletUnderMaintenanceResponse> = z.object({
+  isWalletUnderMaintenance: z.number(),
+});
+
 /**
  * Fetches the status of the Hermez network
  * @returns {Promise<Number>} - Network status of the Hermez Network
  */
 function getNetworkStatus(): Promise<number> {
-  return axios
-    .get(`${baseApiUrl}/network-status.json`)
-    .then((res) => res.data.isWalletUnderMaintenance);
+  return axios.get(`${baseApiUrl}/network-status.json`).then((res) => {
+    const parsedResponse = walletUnderMaintenanceResponseParser.safeParse(res.data);
+    if (parsedResponse.success) {
+      return parsedResponse.data.isWalletUnderMaintenance;
+    } else {
+      return Promise.reject(parsedResponse.error.message);
```

Comment:
My idea here instead of `rejecting` is to return a custom `Promise` to avoid catching errors from the outside. Take a look to this function I wrote in Privado in the last days and let me know what you think: https://github.com/privadonetwork/privado-wallet/blob/main/src/persistence/api.ts#L121 Would be great to discuss different possibilities here.

Reply:
So you're basically wrapping on a box an operation that can either go well or fail, right?

Reply:
Hahahahahaha. Something like a mo... Yeah!

Reply:
Don't say it!